WoW-Realm-Tracker/Controllers/RealmsController.swift

```diff
@@ -1,14 +1,17 @@
+import Swish
+
 public struct RealmsController {
     let realmsDelegate: RealmsDelegate
-    let client = RealmsClient()
+    let client = APIClient()
 
     public init(realmsDelegate: RealmsDelegate) {
         self.realmsDelegate = realmsDelegate
     }
 
     public func retrieveRealms() {
-        client.realmStatus() { realms in
-            self.realmsDelegate.receivedRealms(realms)
+        let request = RealmsRequest()
+        client.performRequest(request) { response in
+            self.realmsDelegate.receivedRealms(response)
```

Comment:
you could actually simplify this and remove the closure altogether, doing `client.performRequest(request, completionHandler: realmsDelegate.recievedRealms)`